main.c

```diff
@@ -303,84 +67,47 @@ int main(void)
+    psa_key_slot_number_t atecc608a_key_slot_psa = 1;
+#endif
 
-    printf("**********************************************\n");
     read_serial_number();
 
-    printf("**********************************************\n");
     hash_sha256(hash_input1, sizeof(hash_input1) - 1,
                 sha256_expected_hash1, sizeof(sha256_expected_hash1));
 
-    printf("**********************************************\n");
     hash_sha256(hash_input2, sizeof(hash_input2) - 1,
                 sha256_expected_hash2, sizeof(sha256_expected_hash2));
 
     /* XXX Call this before or after registering SE? */
     status = psa_crypto_init();
     ASSERT_STATUS(status, PSA_SUCCESS);
-
+#ifdef REGISTER_SE
     /* Register SE */
     status = psa_register_secure_element(atecc608a_drv_info);
     ASSERT_STATUS(status, PSA_SUCCESS);
 
-    /* TODO Register some pre-provisioned SE slots */
-    //status = psa_register_se_slot(
-    //    psa_key_id_t id,
-    //    psa_key_slot_number_t slot,
-    //    psa_key_lifetime_t lifetime,
-    //    psa_key_type_t key_type,
-    //    size_t size,
-    //    uint8_t occupied,
-    //    int32_t owner);
-
-    /* TODO Program the device with some configuration in order to enable it to
-     * sign something? Suggest to copy Azim's for testing configuration, if
-     * needed. Read entire datasheet to find out (cover to cover). */
-
-    /* TODO Make a signature */
-    /* Come up with a test vector. Signature is randomized, so we need to
-     * verify with software. */
-#if USE_SE
-    status = atecc608a_asymmetric_sign(
-        atecc608a_key_slot, alg, &hash, sizeof(hash),
-        signature, sizeof(signature), &signature_length);
-    ASSERT_STATUS(status, PSA_SUCCESS);
-#else
-    /*
-     * Generate a volatile keypair in a volatile key slot.
-     */
-    {
-
-    status = psa_allocate_key(&sign_handle);
-    ASSERT_STATUS(status, PSA_SUCCESS);
+    /* Register first slot */
+    status = psa_register_se_slot(atecc608a_key_slot_psa, atecc608a_key_slot_device,
+                                  PSA_ATECC608A_LIFETIME, key_type,
+                                  pubkey_size, 1, -1);
 
-    psa_key_policy_set_usage(&policy, PSA_KEY_USAGE_SIGN, alg);
-    status = psa_set_key_policy(sign_handle, &policy);
     ASSERT_STATUS(status, PSA_SUCCESS);
+#endif
 
-    status = psa_generate_key(sign_handle, keypair_type, key_bits, NULL, 0);
-    ASSERT_STATUS(status, PSA_SUCCESS);
+    /* Verify that the device has a locked config before doing anything */
+    ASSERT_STATUS(checkLockedZones(), ATCA_CONFIG_ZONE_LOCKED);
 
-    /* Sign a hash. */
-    status = psa_asymmetric_sign(
-        sign_handle, alg, hash, sizeof(hash),
-        signature, sizeof(signature), &signature_length);
+    /* Call the SE functions directly for now */
+    status = atecc608a_export_key(atecc608a_key_slot_device, pubkey,
+                                  sizeof(pubkey), &pubkey_len);
     ASSERT_STATUS(status, PSA_SUCCESS);
 
-    printf("derp\n");
```

Comment:
I don't think we need "derp" in the history. It's just playin' around and not serious.